src/core/server/elasticsearch/elasticsearch_service.ts

```diff
@@ -53,12 +53,16 @@ interface SetupDeps {
 }
 
 /** @internal */
-export class ElasticsearchService implements CoreService<InternalElasticsearchServiceSetup> {
+export class ElasticsearchService
+  implements CoreService<InternalElasticsearchServiceSetup, ElasticsearchServiceStart> {
   private readonly log: Logger;
   private readonly config$: Observable<ElasticsearchConfig>;
   private subscription: Subscription | undefined;
   private stop$ = new Subject();
   private kibanaVersion: string;
+  createClient: InternalElasticsearchServiceSetup['createClient'] | undefined;
+  dataClient: InternalElasticsearchServiceSetup['dataClient'] | undefined;
+  adminClient: InternalElasticsearchServiceSetup['adminClient'] | undefined;
```

Comment:
- Are the fields not `private` for a reason?
- clients should probably be typed as `IClusterClient` intead of using internal setup type reference?

Reply:
good catch, I was relying on VSCode too much.